Add guide for custom connector secret provider with helm installation (#4848)
camunda/camunda-docs: docs/self-managed/connectors-deployment/connectors-configuration.md

provider implementation. Add this JAR to the runtime environment, depending on your deployment setup.
Your secret provider will serve secrets as implemented.

> To use with [Camunda Helm charts](https://artifacthub.io/packages/helm/camunda/camunda-platform), you can build an init container to create a volume with your secret provider and mount it into the Connectors pod. For example, use the following file as input for your Helm install command:

Reviewer: ```suggestion
To use this JAR with [Camunda Helm charts](https://artifacthub.io/packages/helm/camunda/camunda-platform), build an [init container](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/) to create a volume with your secret provider, and mount it into the Connectors pod. 

For example, use the following file as input for your `helm install` command:
```

Small suggestion; it could be nice to link to the supporting kubernetes docs here, and be a little more direct. If "Helm install command" is referring to `helm install`, the inline code formatting might be more descriptive.

```bash
connectors:
  extraVolumes:
    - name: workdir
      emptyDir: {}
  extraVolumeMounts:
    # Mount the secret provider
    # The connectors pod will pick up the secret provider from /opt/app during startup
    - name: workdir
      mountPath: /opt/app/file-secret-provider-2.1.2.jar
      subPath: file-secret-provider-2.1.2.jar
    # Mount the secret
    - name: workdir
      mountPath: /opt/app/mysecret.txt
      subPath: mysecret.txt
  initContainers:
    - name: install
      image: busybox:1.36.1
      command: ["sh", "-c"]
      args:
        # Download a the custom secret provider into the volume
        # As this secret provider works with files, a file with a secret is created as well
        - |
          wget -O /work-dir/file-secret-provider-2.1.2.jar https://artifacts.camunda.com/artifactory/camunda-consulting/com/camunda/consulting/connector/file-secret-provider/2.1.2/file-secret-provider-2.1.2.jar
          touch /work-dir/mysecret.txt

>           echo -n "http://info.cern.ch/" >> /work-dir/mysecret.txt

Reviewer 2: @xevien96 Maybe we can remove the touch/echo part as this is specfic to this secret provider. WDYT?

Author: Sure, it would make the file even cleaner and one could drop in their own secret provider without any other changes
